lib/services/jira.rb

```diff
@@ -102,14 +102,16 @@ def search_issues(params)
   end
 
   def issue_type_by_id(id)
-    raise AhaService::RemoteError, "Integration has not been configured" if meta_data.projects.nil?
     project = meta_data.projects.find {|project| project[:key] == data.project }
-    raise AhaService::RemoteError, "Integration has not been configured, can't find project '#{data.project}'" if project.nil?
-    raise AhaService::RemoteError, "Integration has not been configured, project data not loaded" if meta_data.issue_type_sets.nil?
     issue_types = meta_data.issue_type_sets[project.issue_types]
     issue_type = issue_types.find {|type| type.id.to_s == id.to_s }
-    raise AhaService::RemoteError, "Integration needs to be reconfigured, issue types have changed, can't find issue type '#{id}'" if issue_type.nil?
-    issue_type
+  rescue
+    if meta_data.projects.nil? then raise AhaService::RemoteError, "Integration has not been configured" end
+    if project.nil? then raise AhaService::RemoteError, "Integration has not been configured, can't find project '#{data.project}'" end
+    if meta_data.issue_type_sets.nil? or project.issue_types.nil? or issue_types.nil? then raise AhaService::RemoteError, "Integration has not been configured, project data not loaded" end
+    raise # re-raise original error
+  else
```

Comment:
The alternate you propose feels equally awkward to me, since reading through the happy path, you notice the final line containing a `raise`, and have to read through the entire list of non-happy-path rescues to understand that it isn't actually `rescued`.

The problem is the 3 rescued errors are meant to convert runtime method errors into meaningful ones, whereas the 4th is more of a pre-emptive validation assertion. In the original code they were all of the 4th type, which I'm now thinking was easier to follow the error paths, if not the happy path.

Thoughts on reverting to that old style?